src/sage/arith/misc.py

```diff
@@ -2045,18 +2070,33 @@ def xgcd(a, b):
         sage: xgcd(y^2, a*h*y + b)
         (1, 7*a^2/b^2, (((-h)*a)/b^2)*y + 1/b)
     """
-    try:
+    if b is not None:
+        # xgcd of two elements
+        try:
+            return a.xgcd(b)
+        except AttributeError:
+            a = py_scalar_to_element(a)
+            b = py_scalar_to_element(b)
+        except TypeError:
+            b = py_scalar_to_element(b)
         return a.xgcd(b)
-    except AttributeError:
-        a = py_scalar_to_element(a)
-        b = py_scalar_to_element(b)
-    except TypeError:
-        b = py_scalar_to_element(b)
-    return a.xgcd(b)
+
+    # xgcd of >=2 elements
+    if not isinstance(a, (tuple, list)):
+        raise TypeError("input `a` should be a tuple or a list")
+    if len(a) < 2:
+        raise ValueError("at least two elements should be given")
+    # Compute xgcd recursively
+    res = xgcd(a[0], a[1])
+    for i in range(2, len(a)):
+        g, s, t = xgcd(res[0], a[i])
+        res = [g] + [s * c for c in res[1:]] + [t]
```

Comment:
Just nitpicking here now, but I believe that this can be simplified doing something like that.

```suggestion
    if len(a) == 0:
        return (ZZ(0),)
    a = Sequence(a)
    res = [a.universe().zero()]
    for b in a:
        g, s, t = xgcd(res[0], b)
        res = [g] + [s * c for c in res[1:]] + [t]
```

Reply:
The `Sequence` is what `gcd` also does to make sure that everything lives in the same parent. Otherwise, I am not sure if you could end up in trouble if the elements live in completely different parents. (It's probably fine though, I haven't thought about it too hard.)

Reply:
(Feel free to do things like that or don't, it's just a suggestion.)

Reply:
This looks better. Just for clarification, the return type would be still tuple, right?

Reply:
Yes, tuple is the correct return type. The return type should be the same as it was before.

Reply:
How should I type check `a`? (Or just do not, since `a = Sequence(...)` line eventually gives an error if input cannot be transformed into `Sequence`?)

Reply:
I wouldn't type check. In Python we usually go for duck-typing. Though it has some weird effects:
```
sage: gcd(2 + 2*I)
2
```

Then again, the lack of type checking means that you can put in an iterator:
```
sage: gcd(range(1000))
1
```